src/js/structs/ServiceTree.js

```diff
@@ -115,16 +115,34 @@ module.exports = class ServiceTree extends Tree {
       }
 
       if (filter.health != null && filter.health.length !== 0) {
-        services = services.filter(function (service) {
```

Comment:
@kennyt could you explain a bit why each one of the changes on this file were made?